File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/TouchEvent/ColorUserControl.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace TouchEvent
    Partial Public Class ColorUserControl
        Inherits UserControl
#Region "Filed"

        Private _color As Color
        Private _isNormalSize As Boolean = True

#End Region

#Region "Init"

        Public Sub New()
            InitializeComponent()
            For Each button As Button In tableLayoutPanel1.Controls
                AddHandler button.Click, AddressOf button_Click
            Next
        End Sub

#End Region

#Region "Property"

        Public ReadOnly Property Color() As Color
            Get
                Return _color
            End Get
        End Property

        Public Property IsNormalSize() As Boolean
            Get
                Return _isNormalSize
            End Get
            Set(value As Boolean)
                If _isNormalSize <> value Then
                    _isNormalSize = value
                    If _isNormalSize Then
                        tableLayoutPanel1.Size = New Size(200, 150)
                        Me.Size = New Size(200, 150)
                    Else
                        Me.Size = New Size(400, 300)
                        tableLayoutPanel1.Size = New Size(400, 300)
                    End If
                End If
            End Set
        End Property

#End Region

#Region "Methods and Events"

        Public Event ValueChanged As EventHandler(Of EventArgs)

        Protected Overridable Sub OnValueChanged(e As EventArgs)
            RaiseEvent ValueChanged(Me, e)
        End Sub

        Private Sub button_Click(sender As Object, e As EventArgs)
            _color = TryCast(sender, Button).BackColor
            OnValueChanged(EventArgs.Empty)
        End Sub

#End Region
    End Class
End Namespace

File: Next/TouchToolkit/CS/TouchToolkitExplorer/TouchEvent/ColorUserControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace TouchToolkitExplorer.TouchEvent
{
    public partial class ColorUserControl : UserControl
    {
        #region Filed

        private Color _color;
        private bool _isNormalSize = true;

        #endregion

        #region Init

        public ColorUserControl()
        {
            InitializeComponent();
            foreach (Button button in tableLayoutPanel1.Controls)
            {
                button.Click += button_Click;
            }
        }

        #endregion

        #region Property

        public Color Color
        {
            get
            {
                return _color;
            }
        }

        public bool IsNormalSize
        {
            get
            {
                return _isNormalSize;
            }
            set
            {
                if (_isNormalSize != value)
                {
                    _isNormalSize = value;
                    if (_isNormalSize)
                    {
                        tableLayoutPanel1.Size = new Size(200, 150);
                        this.Size = new Size(200, 150);
                    }
                    else
                    {
                        this.Size = new Size(400, 300);
                        tableLayoutPanel1.Size = new Size(400, 300);
                    }
                }
            }
        }

        #endregion

        #region Methods and Events

        public event EventHandler<EventArgs> ValueChanged;

        protected virtual void OnValueChanged(EventArgs e)
        {
            if (this.ValueChanged != null)
            {
                this.ValueChanged(this, e);
            }
        }

        void button_Click(object sender, EventArgs e)
        {
            _color = (sender as Button).BackColor;
            OnValueChanged(EventArgs.Empty);
        }

        #endregion
    }
}
